new/change/write-main-changelog.ts

```diff
@@ -0,0 +1,32 @@
+import type { Module } from '../types';
+import { logger } from '../helpers/logger';
+import { join } from 'path';
+
+import fs from 'fs';
+
+const { error, log } = logger('[change] write main changelog');
+
+export const writeMainChangelog: Module = (env) => {
+  const changelogJson = env.changelog;
+  const mainChangeLogDir = env.config!.changelogPath;
+
+  if (!mainChangeLogDir) {
+    const errorMessage = 'changelog path is undefined!';
+    error(errorMessage);
+  }
+
+  const mainChangeLogFile = join(
+    `${env.config!.changelogPath}`,
```

Comment:
since this is an interpolated string, we don't need join here I guess? a simple `${env.config!.changelogPath}changelog.json` would suffice?